src/main/java/life/qbic/io/commandline/UploadDatasetCommand.java

```diff
@@ -54,10 +63,19 @@ public void run() {
       System.out.println();
       System.out.println("Parameters verified, uploading dataset...");
       System.out.println();
-      DataSetPermId result = openbis.registerDataset(Path.of(dataPath), experimentID, parents);
-      System.out.printf("Dataset %s was successfully created%n", result.getPermId());
+      if(attachToExperiment) {
+        DataSetPermId result = openbis.registerDatasetForExperiment(Path.of(dataPath), objectID, parents);
+        System.out.printf("Dataset %s was successfully created%n", result.getPermId());
+      } else {
```

Comment:
NIT: I think it would help to have 2 different output messages here, since one result attaches the dataset to an experiment while the other registers it to a sample